### PR TITLE
Add Pre-warm TLS Tunnel

### PR DESCRIPTION
**Pull request by masaori335**
The main change of #7241. For now, this has only the new code for the pre-warming TLS tunnel.
Definitely, I need to work more on rebasing my development branch and cleaning up.

### CODE REVIEW

**Comment by ezelkow1** [comment]
[approve ci]

**Comment by bryancall** [comment]
@shinrich is going to review it.
@masaori335 is going to check to see if internal crash fix is in this PR.

**Comment by masaori335** [comment]
[approve ci autest]

**Comment by masaori335** [comment]
Rebased on the latest master. This has #8323 changes to avoid conflicts for now. ( I'll rebase when it's merged )

**Comment by bryancall** [comment]
@ffcai is going to take a look at this

**Review by duke8253 — APPROVED** [review]

File: iocore/net/YamlSNIConfig.h [hunk 1]
```diff
@@ -77,6 +85,14 @@ struct YamlSNIConfig {
     unsigned long protocol_mask;
     std::vector<int> tunnel_alpn{};
 
+    bool tunnel_prewarm_srv                  = false;
+    uint32_t tunnel_prewarm_min              = 0;
+    int32_t tunnel_prewarm_max               = -1;
+    double tunnel_prewarm_rate               = 1.0;
+    uint32_t tunnel_prewarm_connect_timeout  = 0;
```
[the remark this answers is not in the document]
Comment (masaori335) [reply iocore/net/YamlSNIConfig.h:92]: Right

**Comment by masaori335** [comment]
@duke8253 Could you take a look again? I just rebased on the latest master.

**Review by duke8253 — APPROVED** [review]
Looks good to me.

**Pull request merged**